IdentityCore/tests/MSIDLRUCacheTest.m

```diff
@@ -0,0 +1,386 @@
+                                                                                                         throttleType:cacheKey
+                                                                                                     throttleDuration:100];
+            
+            [customLRUCache setObject:throttleCacheRecord
+                               forKey:cacheKey
+                                error:&subError];
+        }
+        [expectation2 fulfill];
+    });
+    
+
+    [self waitForExpectations:expectationsAdd timeout:20];
+    XCTAssertEqual(customLRUCache.numCacheRecords,100);
+
+
+    dispatch_async(parentQ1, ^{
+        for (int i = 0; i < 50; i++)
+        {
+            NSString *cacheKey = [NSString stringWithFormat:@"%i", i]; //0-49
+            NSString *cacheKeyFromOtherQueue = [NSString stringWithFormat:@"%i", (i + 50)]; //50-99
+            if (i % 2)
+            {
+
+                [customLRUCache removeObjectForKey:cacheKey
+                                             error:&subError];
+                XCTAssertNil(subError);
+
+                MSIDThrottlingCacheRecord *throttleCacheRecord = [[MSIDThrottlingCacheRecord alloc] initWithErrorResponse:nil
+                                                                                                             throttleType:cacheKeyFromOtherQueue
+                                                                                                         throttleDuration:100];
+
+                [customLRUCache setObject:throttleCacheRecord
+                                   forKey:cacheKeyFromOtherQueue
+                                    error:&subError];
+            
+                XCTAssertNil(subError);
+            }
+        }
+        [expectation3 fulfill];
+    });
+
+    dispatch_async(parentQ2, ^{
+        for (int i = 50; i < 100; i++)
+        {
+            NSString *cacheKey = [NSString stringWithFormat:@"%i", i]; //50-99
+            NSString *cacheKeyFromOtherQueue = [NSString stringWithFormat:@"%i", (i - 50)]; //0-49
+            if (i % 2)
+            {
+                [customLRUCache removeObjectForKey:cacheKey
+                                             error:&subError];
+                XCTAssertNil(subError);
+                MSIDThrottlingCacheRecord *throttleCacheRecord = [[MSIDThrottlingCacheRecord alloc] initWithErrorResponse:nil
+                                                                                                             throttleType:cacheKeyFromOtherQueue
+                                                                                                         throttleDuration:100];
+
+                [customLRUCache setObject:throttleCacheRecord
+                                   forKey:cacheKeyFromOtherQueue
+                                    error:&subError];
+                XCTAssertNil(subError);
+            }
+        }
+
+        [expectation4 fulfill];
+    });
+
+    [expectation4 fulfill];
```

Comment:
There was a minor bug in the last unit test, where expectation fulfill was outside of the async block. Sorry for the inconvenience, @oldalton  Could you please pull & merge with the latest dev again? I will then approve it for you.

Reply:
@NerevarineRule, can you submit new PR against this release branch with the needed changes?
We shouldn't pull dev into release according to the gitflow. Thanks!